### PR TITLE
Recommend setuid for mtr

### PR DESCRIPTION
**Pull request by charliesome**
mtr drops privileges immediately after starting up and opening raw sockets:

https://github.com/traviscross/mtr/blob/7dbc602cd558d8a0835a323511e4997b8d5d34c1/mtr.c#L562-L579

It should be fairly safe to setuid.

cc @mikemcquaid

### CODE REVIEW

File: Library/Formula/mtr.rb [hunk 1]
```diff
@@ -36,8 +36,15 @@ def install
   end
 
   def caveats; <<-EOS.undent
-    mtr requires root privileges so you will need to run `sudo mtr`.
-    You should be certain that you trust any software you grant root privileges.
+    mtr requires superuser privileges for raw sockets access. You can either
+    run the program via `sudo`, or change its ownership to root and set the
+    setuid bit:
+
+      sudo chown root:wheel #{sbin}/mtr
```
Comment (bfontaine) [inline Library/Formula/mtr.rb]: Wouldn’t this break `brew rm mtr`?

Comment (DomT4) [inline Library/Formula/mtr.rb]: Yes, almost certainly. I think the current caveat is fine, personally.

Comment (charliesome) [inline Library/Formula/mtr.rb]: Deleting a file only requires write permission to the containing directory, the permissions on the file itself don't matter.

Comment (DomT4) [inline Library/Formula/mtr.rb]: Ah, yes. Apologies - I misread and thought you were chowing the whole directory, which would cause issues.

**Comment by mikemcquaid** [comment]
@charliesome I'd probably rather not do this not because it isn't safe but because it will encourage other formulae (which aren't safe) to do the same. My concern here is also if there's a bug in `mtr` that stops it dropping privileges or e.g. a buffer overflow which allows code execution then having it be `root` is more concerning.

tl;dr: the people who I trust with `setuid` will already know how to set it.

**Comment by DomT4** [comment]
Going to pass on this per Mike's comment. If you're comfortable setting it yourself though, by all means, feel free. Thanks for the PR!

**Comment by charliesome** [comment]
No worries :sunglasses:

**Pull request closed without merging**